trl/trainer/callbacks.py

```diff
@@ -429,3 +430,64 @@ def on_step_end(self, args, state, control, **kwargs):
 
         # Save the last logged step, so we don't log the same completions multiple times
         self._last_logged_step = state.global_step
+
+
+class MergeModelCallback(TrainerCallback):
+    r"""
+    A [`~transformers.TrainerCallback`] that merges the policy model (the model being trained) with another model based on a merge configuration.
+
+    Usage:
+    ```python
+    pip install trl[mergekit]
+
+    trainer = DPOTrainer(...)
+    from trl.mergekit_utils import MergeConfig
+    from trl import MergeModelCallback
+    config = MergeConfig("ties")
+    merge_callback = MergeModelCallback(config)
+    trainer.add_callback(merge_callback)
+    ```
+
+    Args:
+       merge_config (`MergeConfig`):
+           The configuration used for the merging process.
+    - `merge_at_every_checkpoint` =: `bool`, optional, default=`False`. Merges the model at every checkpoint instead of just at the end.
+    - `push_to_hub`: `bool`, optional, default=`False`. Pushes the merged model to Hugging Face hub.
+    """
+
+    def __init__(self, merge_config, push_to_hub=False, merge_at_every_checkpoint=False):
+        self.push_to_hub = push_to_hub
```

Comment:
push_to_hub should be gotten from the training args  I think

Reply:
push_to_hub from training args pushes the trained model 
this one pushes the merged model